lib/fs.js

```diff
@@ -2479,3 +2479,452 @@ Object.defineProperty(fs, 'SyncWriteStream', {
   set: internalUtil.deprecate((val) => { SyncWriteStream = val; },
                               'fs.SyncWriteStream is deprecated.', 'DEP0061')
 });
+
+
+// The Promises API
+
+async function* readEverything(fd) {
+  let eof = false;
+  let position = 0;
+  const buffer = Buffer.allocUnsafe(kReadFileBufferLength);
+  while (!eof) {
+    const { length } =
+      await fs.promises.read(fd, buffer, 0, buffer.length, position);
```

Comment:
I’d drop the `, position` if it isn’t necessary… without it, this should work on un-seekable files (e.g. TTYs, pipes) as well?

Reply:
Yeah, had that on my list to work through